app/controllers/friendships_controller.rb

```diff
@@ -2,15 +2,24 @@ class FriendshipsController < ApplicationController
   def create
     friend = User.find_by(email: params[:email])
     if friend.present?
-      friendship = Friendship.new(user_id: current_user.id, friend_id: friend.id)
-      if friendship.save
-        flash[:sucess] = 'Friend successfully added!'
-      else
-        flash[:error] = 'Friendship not successfully created.'
-      end
+      validate_friendship(friend)
     else
       flash[:error] = "Unable to add friend. User with email #{params[:email]} not found."
     end
     redirect_to dashboard_path
   end
+
+  private
+
+  def validate_friendship(friend)
+    if current_user.add_friend(friend)
+      flash[:sucess] = 'Friend successfully added!'
+    elsif friend.id == current_user.id
+      flash[:error] = "Error: This is just sad. You can\'t add yourself as a friend."
+    elsif current_user.friends.include?(friend)
+      flash[:error] = 'Error: You are already friends with this user.'
+    else
+      flash[:error] = 'Friendship not successfully created.'
+    end
+  end
```

Comment:
I really like this was of validating a friend! Great Job this looks really good!